src/engine/Engine.cxx

```diff
@@ -23,7 +23,7 @@ void Engine::decreaseHeight(const Point &isoCoordinates) const
 
 void Engine::toggleFullScreen() { WindowManager::instance().toggleFullScreen(); };
 
-void Engine::setTileIDOfNode(const Point &isoCoordinates, const std::string &tileID) const
+void Engine::setTileIDOfNode(const std::vector<Point> &isoCoordinates, const std::string &tileID) const
```

Comment:
Instead of sending a `std::vector` reference, we should use Iterators to be more generic. For example,
```cpp
template <typename Iterator>
void Engine::setTileIDOfNode(Iterator begin, Iterator end, const std::string &tileID) const
{
  static_assert(std::is_same_v<const Point, typename std::iterator_traits<Iterator>::value_type>, "Iterator value must be a const Point");
  map->setTileIDOfNode(begin, end, tileID);
}
```
This would allow us to use this algorithm on other containers (e.g. deques, linked lists, etc)